### ISSUE TITLE
Zed to remember opened terminals by project

### ISSUE THREAD

**Issue by snturk** [zed-industries/zed#11841]
### Check for existing issues

- [X] Completed

### Describe the feature

I work a couple of projects on Zed, so I need Zed to remember the opened terminals when I reopen a project.

Currently, if I open 3 terminals on project X and `cd` on them to different directories, Zed only remembers the number of terminals and displays it on all projects.

**Comment by etiennelacoursiere** [comment]
I would love that to!

In my current workflow with tmux and neovim i switch projects ( sessions ) with `<prefix> s`. This is perfect as i can hop on another project, do work, then come back to my project and still have my session with my neovim open in a pane, my server running in another pane and any other terminal i might have open in that project for various reason. 

I'm not sure if this is possible but what i would love in zed is to do something similar. 

Have a project open ( a ) , with some terminals open that run a server, logged into a ssh session or whatever. 
Switch to a different project ( b )  via the `Recent Project` switcher and selecting with `cmd + enter` to keep the same window.
When i eventually come back to project a, my terminal states are intact, the server is still running in a terminal, the ssh session still open in the other, etc. 

Currently, in Zed i can do this with TMUX still. when i swtich to another project, i attach to tmux and go to this project session, and can have my panes and all. It works, but it would be really great to do it natively without tmux.

**Comment by VimCommando** [comment]
I believe this has been delivered with the new multi-agent release! Projects/worktrees keep their own window and tab sets, including terminals, when switching between them.

**Comment by el-analista** [comment]
no, when I suspend the macbook and come back the projects are in the sidebar but all the terminal agents(using tmux) have dissapeared so I have to open again tmux attach and remember which session I was using
